torchao/quantization/subclass.py

```diff
@@ -927,62 +893,170 @@ def __torch_dispatch__(cls, func, types, args, kwargs):
+
+    if func == aten.addmm.default:
+        assert args[1].shape[-1] == args[2].shape[0], (
+            f"need mat1 shape: {args[1].shape} final"
+            f"dim to match mat2 shape: {args[2].shape} first dim "
+        )
+        input_tensor, weight_qtensor, bias = (
+            args[1],
+            args[2],
+            args[0],
+        )
+    else:
+        assert args[0].shape[-1] == args[1].shape[0], (
+            f"need mat1 shape: {args[0].shape} final dim"
+            f"to match mat2 shape: {args[1].shape} first dim"
+        )
+        input_tensor, weight_qtensor, bias = (
+            args[0],
+            args[1],
+            None if len(args) == 2 else args[2],
+        )
+    weight_tensor = weight_qtensor.dequantize()
+    return func(input_tensor, weight_tensor, bias)
 
-        if func is aten.detach.default:
-            return return_and_correct_aliasing(
-                func, args, kwargs, args[0]._apply_fn_to_data(torch.detach)
-            )
+@implements_aqt_aten_ops([aten.detach.default])
+def detach(func, *args, **kwargs):
+    return return_and_correct_aliasing(
+        func, args, kwargs, args[0]._apply_fn_to_data(torch.detach)
+    )
 
-        if func is aten.clone.default:
-            return return_and_correct_aliasing(
-                func, args, kwargs, args[0]._apply_fn_to_data(torch.clone)
-            )
 
-        if func is aten.t.default:
-            # TODO: need to implement this
-            # args[0].transposed = not args[0].transposed
-            # new = args[0]._change_shape(args[0].shape[::-1])
-            # return return_and_correct_aliasing(func, args, kwargs, new)
-            raise Exception("transpose not implemented yet")
+@implements_aqt_aten_ops([aten.clone.default])
+def clone(func, *args, **kwargs):
+    return return_and_correct_aliasing(
+        func, args, kwargs, args[0]._apply_fn_to_data(torch.clone)
+    )
 
-        if func is aten._to_copy.default:
-            return return_and_correct_aliasing(
-                func,
-                args,
-                kwargs,
-                args[0].to(*args[1:], **kwargs)._apply_fn_to_data(torch.clone),
-            )
 
-        raise NotImplementedError(
-            f"AffineQuantizedTensor dispatch: attempting to run {func}, this is not supported"
-        )
+@implements_aqt_aten_ops([aten._to_copy.default])
+def _to_copy(func, *args, **kwargs):
+    return return_and_correct_aliasing(
+        func,
+        args,
+        kwargs,
+        args[0].to(*args[1:], **kwargs)._apply_fn_to_data(torch.clone),
+    )
+
+@implements_aqt_aten_ops([aten.t.default])
+def t(func, *args, **kwargs):
+    # TODO: need to implement this
+    # args[0].transposed = not args[0].transposed
+    # new = args[0]._change_shape(args[0].shape[::-1])
+    # return return_and_correct_aliasing(func, args, kwargs, new)
+    raise Exception("transpose not implemented yet")
```

Comment:
I think you could even `return NotImplemented`, which will result in a more standardized stack trace. Give it a try if you like.